app/src/main/java/com/duckduckgo/app/browser/DuckDuckGoRequestRewriter.kt

```diff
@@ -63,14 +64,25 @@ class DuckDuckGoRequestRewriter(
     /**
      * Applies cohort (atb) https://duck.co/help/privacy/atb and
      * source (t) https://duck.co/help/privacy/t params to url
+     * Also keeps current Vertical selection (Images, News, Videos, etc...)
      */
-    override fun addCustomQueryParams(builder: Uri.Builder) {
+    override fun addCustomQueryParams(currentUrl: String, builder: Uri.Builder) {
         val atb = statisticsStore.atb
         if (atb != null) {
             builder.appendQueryParameter(ParamKey.ATB, atb.formatWithVariant(variantManager.getVariant()))
         }
 
         val sourceValue = if (appReferrerDataStore.installedFromEuAuction) ParamValue.SOURCE_EU_AUCTION else ParamValue.SOURCE
+
+        if (variantManager.getVariant().hasFeature(VariantManager.VariantFeature.SerpHeaderRemoval)){
+            builder.appendQueryParameter(ParamKey.HIDE_SERP, ParamValue.HIDE_SERP)
+        }
+
+        if (duckDuckGoUrlDetector.isDuckDuckGoVerticalUrl(currentUrl)){
```

Comment:
this fixes a problem that occurs in the released version, requerying when a vertical is selected (Images, News, etc...) from the native toolbar send the user to the `All` vertical instead of staying in the previously selected one